File: palindrom.cpp
#include<iostream>
#include<string>

int main()
{
	int n,m;
	std::string word;

	std::cout << "Podaj wyraz, ktory chcesz sprawdzic: ";
	std::cin >> wyraz;

	m = word.length()-1;

	for (n = 0; n<m; n++)
	{
		if (word[n] != word[m-n])
		break;
	}
	if (n < m)
		std::cout << "Podany wyraz nie jest palindromem" <<std::endl;
	else
		std::cout << "Podany wyraz jest palindromem" <<std::endl;

return 0;

}

File: palindrom.py


przyklad1 = "dupa"
przyklad2 = "kajak"


def czy_palindrom(slowo):
	if slowo == slowo[::-1]:
		print("slowo ", slowo, "jest palindromem")
	else:
		print("slowo", slowo, "nie jest palindromem")



czy_palindrom(przyklad1)
czy_palindrom(przyklad2)